config/initializers/schema_validator.rb

```diff
@@ -0,0 +1,241 @@
+# Schema validation for configuration files
+# This provides basic validation of the configuration structure
+
+module SchemaValidator
+  # Validate the entire configuration structure
+  def self.validate_config(config)
+    errors = []
+
+    # Check if config is nil or not a hash-like object
+    return ['Configuration is empty or not properly loaded'] if config.nil?
+
+    # Get if object responds to hash-like methods
+    has_key_method = config.respond_to?(:has_key?) || config.respond_to?(:key?)
+
+    # Choose validation approach based on object type
+    if has_key_method
+      # Hash-like validation (Settings object or Hash)
+
+      # Check for required keys using appropriate method
+      check_key = config.respond_to?(:has_key?) ? :has_key? : :key?
+      required_keys = %w{default_org default_repo default_branches default_days}
+
+      # Only check for organizations if we're not in a test environment
+      # Only require organizations in non-test mode
+      unless ENV['CHEF_OSS_STATS_TEST_MODE'] == 'true'
+        required_keys << 'organizations'
+      end
+
+      missing_keys = required_keys.select { |key| !config.send(check_key, key) }
+      unless missing_keys.empty?
+        errors << 'Missing required configuration keys: ' \
+                 "#{missing_keys.join(', ')}"
+      end
+
+      # Type validation for core keys
+      begin
+        # Validate default_org is a string if it exists
+        if config_has_key?(config, 'default_org') &&
+           !config_value(config, 'default_org').is_a?(String)
+          errors << 'default_org must be a string'
+        end
+
+        # Validate default_repo is a string if it exists
+        if config_has_key?(config, 'default_repo') &&
+           !config_value(config, 'default_repo').is_a?(String)
+          errors << 'default_repo must be a string'
+        end
+
+        # Validate default_branches is an array if it exists
+        if config_has_key?(config, 'default_branches') &&
+           !config_value(config, 'default_branches').is_a?(Array)
+          errors << 'default_branches must be an array'
+        end
+
+        # Validate default_days is a number if it exists
+        if config_has_key?(config, 'default_days') &&
+           !config_value(config, 'default_days').is_a?(Numeric)
+          errors << 'default_days must be a number'
+        end
```

Comment:
I feel like default_days and default_branches can have their own fallback, and people only need to specify them if they want the default to be different than the built in.